src/kikuchipy/detectors/ebsd_detector.py

```diff
@@ -215,10 +215,11 @@ def __init__(
         """
         self.shape = shape
         self.px_size = float(px_size)
-        self.binning: float = float(binning)
-        self.tilt = float(tilt)
-        self.azimuthal = float(azimuthal)
+        self.binning = float(binning)
+        self._tilt = float(tilt)
+        self._azimuthal = float(azimuthal)
         self.sample_tilt = float(sample_tilt)
+        self._euler = np.array([self.azimuthal, 90.0 + self.tilt, 0.0]).astype(float)
```

Comment:
There are plenty of examples where the third detector Euler angle is non-zero. At least for the Oxford Instruments software, the detector orientation is stored in the h5oina as Euler angles. I think there is a case for being able to set the detector Euler angles directly e.g. when using the `kp.load()` function to read a dataset. On the other hand, if you prefer to define the angles separately, it would be possible to define a third angle, something like `EBSDDetector.twist`, which would be equivalent to `EBSDDectector.euler[2]`.

Reply:
You have to educate me here, @tgwoodcock. The camera tilt theta and azimuthal angles omega have simple enough meanings, as seen in (b) and (d) below. What does the twist look like, in terms of our reference frame figure?

![bilde](https://github.com/user-attachments/assets/ad64239d-a072-4d4e-b838-45211ee7eb53)

Perhaps @IMBalENce has thought about this too?

Reply:
The third detector Euler angle is a rotation about the detector normal (Z_d) after having performed the first two rotations. It's usually small but non-zero, at least for OxInst systems, where the orientation of the detector is expressed as detector Euler angles in the measurement metadata. This third detector Euler angle is the `twist` I would suggest introducing, if we don't want to set the detector Euler angles directly. The `twist` could default to zero and if needed, users can set it during the instantiation and then the dependent property `euler` will be correct.

Reply:
Thank you for clarifying. I'm convinced. Now that we're refactoring the angles, do you have a suggestion for better naming of them?

Reply:
We could maybe think about calling the three detector angles roll, pitch and yaw, but on the other hand, I think the current naming is ok too. I'll start updating my branch with the current names and we can change them later if needed.

Reply:
I've seen those three being used for explaining Euler angles, but never in the context of crystals. However, many software have "Camera tilt" (EMsoft, Bruker, etc.), and at least EMsoft has the "azimuthal" camera angle as well. This is the reason why these two angles have been called this in kikuchipy. So, having camera tilt, azimuthal, and twist I think should be fine. Go with twist!